File: scripts/java/AESUtil.py
#!/usr/bin/env python3
"""AES/ECB/PKCS5Padding utility matching the original Java helper."""

from __future__ import annotations

import base64
import json
import secrets
from dataclasses import dataclass
from datetime import date

from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes


SECRET = "Btm/Cb6N6glbcOEvjV8qGnyQELjWFUkD"
AES_BLOCK_SIZE = 16


@dataclass
class AESSecret:
    appKey: str
    appSecret: str


def _key_bytes(key: str) -> bytes:
    key_data = key.encode()
    if len(key_data) not in (16, 24, 32):
        raise ValueError("AES key must be 16, 24, or 32 bytes after UTF-8 encoding")
    return key_data


def _pkcs5_pad(data: bytes) -> bytes:
    # AES 的块大小为 16 字节；Java PKCS5Padding 在 AES 场景下等价于 PKCS7。
    padding_len = AES_BLOCK_SIZE - (len(data) % AES_BLOCK_SIZE)
    return data + bytes([padding_len]) * padding_len


def _pkcs5_unpad(data: bytes) -> bytes:
    if not data:
        raise ValueError("decrypted data is empty")
    padding_len = data[-1]
    if padding_len < 1 or padding_len > AES_BLOCK_SIZE:
        raise ValueError("invalid PKCS5 padding")
    if data[-padding_len:] != bytes([padding_len]) * padding_len:
        raise ValueError("invalid PKCS5 padding")
    return data[:-padding_len]


def aes_encrypt(text: str, key: str) -> str:
    """Encrypt text with AES/ECB/PKCS5Padding and return Base64 ciphertext."""
    cipher = Cipher(algorithms.AES(_key_bytes(key)), modes.ECB())
    encryptor = cipher.encryptor()
    encrypted = encryptor.update(_pkcs5_pad(text.encode())) + encryptor.finalize()
    return base64.b64encode(encrypted).decode()


def aes_decrypt(base64_encrypted: str, key: str) -> str:
    """Decrypt Base64 ciphertext created by aes_encrypt."""
    cipher = Cipher(algorithms.AES(_key_bytes(key)), modes.ECB())
    decryptor = cipher.decryptor()
    encrypted = base64.b64decode(base64_encrypted)
    decrypted = decryptor.update(encrypted) + decryptor.finalize()
    return _pkcs5_unpad(decrypted).decode()


def get_app_key_and_secret() -> AESSecret:
    """Generate an appKey/appSecret pair in the same Base64 shape as Java."""
    app_key = base64.b64encode(secrets.token_bytes(16)).decode()
    app_secret = base64.b64encode(secrets.token_bytes(24)).decode()
    return AESSecret(appKey=app_key, appSecret=app_secret)


# 保留 Java 原方法名，便于直接替换旧调用。
def AESEncrypt(text: str, key: str) -> str:
    return aes_encrypt(text, key)


def AESDecrypt(base64Encrypted: str, key: str) -> str:
    return aes_decrypt(base64Encrypted, key)


def getAPPKeyAndSecret() -> AESSecret:
    return get_app_key_and_secret()


def main() -> None:
    body = {
        "snCode": "111",
        "date": date.today().isoformat(),
    }
    text = json.dumps(body, ensure_ascii=False, separators=(",", ":"))
    print(text)
    encrypted = aes_encrypt(text, SECRET)
    print(encrypted)
    print(aes_decrypt(encrypted, SECRET))


if __name__ == "__main__":
    main()

File: scripts/java/AESUtil.java
package com.hzfj.glasses.utils;

import com.alibaba.fastjson.JSON;
import com.hzfj.glasses.model.dto.AuthCheckBody;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;
import org.springframework.format.annotation.DateTimeFormat;

import javax.crypto.Cipher;
import javax.crypto.KeyGenerator;
import javax.crypto.SecretKey;
import javax.crypto.spec.SecretKeySpec;
import java.security.NoSuchAlgorithmException;
import java.security.SecureRandom;
import java.time.LocalDate;
import java.util.Base64;


public class AESUtil {
    /**
     * AES算法加密
     * @Param:text原文
     * @Param:key密钥
     * */
    public static String AESEncrypt(String text, String key) throws Exception {
        // 创建AES加密算法实例(根据传入指定的秘钥进行加密)
        Cipher cipher = Cipher.getInstance("AES/ECB/PKCS5Padding");
        SecretKeySpec keySpec = new SecretKeySpec(key.getBytes(), "AES");
        // 初始化为加密模式，并将密钥注入到算法中
        cipher.init(Cipher.ENCRYPT_MODE, keySpec);
        // 将传入的文本加密
        byte[] encrypted = cipher.doFinal(text.getBytes());
        //生成密文
        // 将密文进行Base64编码，方便传输
        return Base64.getEncoder().encodeToString(encrypted);
    }

    /**
     * AES算法解密
     * @Param:base64Encrypted密文
     * @Param:key密钥
     * */
    public static String AESDecrypt(String base64Encrypted, String key) throws Exception {
        // 创建AES解密算法实例
        Cipher cipher = Cipher.getInstance("AES/ECB/PKCS5Padding");
        SecretKeySpec keySpec = new SecretKeySpec(key.getBytes(), "AES");
        // 初始化为解密模式，并将密钥注入到算法中
        cipher.init(Cipher.DECRYPT_MODE, keySpec);
        // 将Base64编码的密文解码
        byte[] encrypted = Base64.getDecoder().decode(base64Encrypted);
        // 解密
        byte[] decrypted = cipher.doFinal(encrypted);
        return new String(decrypted);
    }

    public static AESSecret getAPPKeyAndSecret() {
        AESSecret aesSecret = new AESSecret();
        SecretKey secretKey = generateAESKey();
        String appKey = Base64.getEncoder().encodeToString(secretKey.getEncoded());
        SecretKeySpec extend = getSecretKey(appKey);
        String appSecret = Base64.getEncoder().encodeToString(extend.getEncoded());
        aesSecret.setAppKey(appKey);
        aesSecret.setAppSecret(appSecret);
        return aesSecret;
    }

    private static SecretKey generateAESKey() {
        KeyGenerator keyGen;
        try {
            keyGen = KeyGenerator.getInstance("AES");
        } catch (NoSuchAlgorithmException e) {
            throw new RuntimeException(e);
        }
        keyGen.init(128); // AES key size
        return keyGen.generateKey();
    }

    /**
     * 生成加密秘钥
     *
     * @return
     */
    private static SecretKeySpec getSecretKey(String password) {

        //返回生成指定算法密钥生成器的 KeyGenerator 对象
        KeyGenerator kg;
        try {
            kg = KeyGenerator.getInstance("AES");
            SecureRandom secureRandom = SecureRandom.getInstance("SHA1PRNG");
            secureRandom.setSeed(password.getBytes());
            //AES 要求密钥长度为 192
            kg.init(192, secureRandom);
            //生成一个密钥
            SecretKey secretKey = kg.generateKey();
            // 转换为AES专用密钥
            return new SecretKeySpec(secretKey.getEncoded(), "AES");
        } catch (NoSuchAlgorithmException e) {
            throw new RuntimeException(e);
        }
    }

    @Data
    public static class AESSecret {

        public String appKey;

        public String appSecret;
    }

    private static final String SECRET = "Btm/Cb6N6glbcOEvjV8qGnyQELjWFUkD";

    public static void main(String[] args) throws Exception {
//        AESSecret appKeyAndSecret = getAPPKeyAndSecret();
//        System.out.println(appKeyAndSecret.getAppKey());
//        System.out.println(appKeyAndSecret.getAppSecret());

        AuthCheckBody body = new AuthCheckBody();
        body.setSnCode("111");
        body.setDate(LocalDate.now());
        String json = JSON.toJSONString(body);
        System.out.println(json);
        String encrypt = AESEncrypt(json, SECRET);
        System.out.println(encrypt);
        String decrypt = AESDecrypt(encrypt, SECRET);
        System.out.println(decrypt);
    }

}
